### DISCUSSION TITLE
create FSM simulation class to handle simulations

### DISCUSSION THREAD

**Discussion by ArcticSnow** [ArcticSnow/TopoPyScale#87] [Ideas]
The whole FSM simulation part could be bundled into a class to streamline the processing, add consistency in parameters (path, filename, etc):
1. handle files (config, inputs and outputs)
2. convert output to netcdf
3. parallelize simulations for large job
4. include pertubation system
5. and eventually include data assimilation too

**Comment by ealonsogzl** [comment]
You can probably just use/adapt this

https://github.com/ealonsogzl/MuSA/blob/master/modules/internal_class.py

**Reply by ArcticSnow** [reply]
thank you @ealonsogzl . 
Can we plan some time next week for talking about combining MuSa and TopoPyScale?

**Comment by ealonsogzl** [comment]
Sure, we'll have a lot of time :)

I don't see a big problem in doing so, and this issue would be solved. 
Now in MuSA we can insert any model, so the capabilities of Toposcale would be greatly extended.

The only question is to generate a generalist piece of code able to add observations to the Toposcale geometry if you want to DA. But if dealing with observation geometries is too challenging, MuSA can be used as a system to couple Toposcale with other models easily.

**Reply by ArcticSnow** [reply]
Cool! looking forward to hear all about it!

**Comment by joelfiddes** [comment]
great Esteban!  theres for sure a tonne of stuff we could use from MuSA - I particularly like the auto setup of FSM, Im always going mad with different versions!

We also need to keep in mind the different use cases that need to be served - lets discuss in more detail when we meet.
